frontend/src/components/auth/Auth.tsx

```diff
@@ -0,0 +1,22 @@
+/**
+ * Component to manage login/authorisation process
+ */
+
+import React from "react";
+
+import authService from "../../services/auth";
+import useObservable from "../../hooks/observable";
+
+import LoginPage from "./login/Login";
+import RegisterPage from "./register/Register";
+
+/**
+ * Renders login or registration page
+ */
+function AuthPage() {
+  const login = useObservable(authService.loginOrRegister());
+
+  return login ? <LoginPage /> : <RegisterPage />;
```

Comment:
You should use enum instead of boolean and put switch-case statement here. See the comments below for details